library/Icingadb/Redis/VolatileStateResults.php

```diff
@@ -114,13 +124,14 @@ protected function applyRedisUpdates()
         }
     }
 
-    protected function fetchStates(string $key, array $ids): Generator
+    protected function fetchStates(string $key, array $ids, array $keys): Generator
     {
         $results = IcingaRedis::instance()->getConnection()->hmget($key, $ids);
         foreach ($results as $i => $json) {
             if ($json !== null) {
                 $data = json_decode($json, true);
-                $data = array_intersect_key($data, array_flip(VolatileState::$keys));
+                $data = array_merge(array_fill_keys($keys, null), $data);
+                $data = array_intersect_key($data, $keys);
```

Comment:
Where did the array_flip went? Now $data is empty and the redis result is not used??